lib/src/command/deps.dart

```diff
@@ -67,23 +72,59 @@ class DepsCommand extends PubCommand {
     if (argResults['executables']) {
       _outputExecutables();
```

Comment:
I have mixed feelings about this.

We could choose to output different JSON output when either `--dev` and/or `--executable` is given, or we could just ignore (or forbid) these flags when outputting JSON, and always output the same thing.

Since a JSON output could contain a property `"relation": "direct | dev | transitive | transitive-dev"` or something like that, and we leave it up to the reader to parse it.

Didn't we do this for `dart pub outdated --json` ?